### DISCUSSION TITLE
video:shape with letterboxing

### DISCUSSION THREAD

**Discussion by bradh** [stac-extensions/video#6] [General]
One thing that comes up a bit is letterboxing. The concept is around what counts as a row or column. Is it the encoded size, or the "non-fill" pixels?

Here is an example:
![image](https://user-images.githubusercontent.com/174642/153341983-614d8800-9cea-4022-83db-8548ef54cb7a.png)

Note the black pixels down either side.

Its even more common on IR sensors (which rarely line up with standard encoding sizes), as shown in this example:
![image](https://user-images.githubusercontent.com/174642/153342238-e37b10d1-0502-4415-b422-ee5ea9ddfc92.png)

Ideally we'd be able to encode both, but it may not be possible to easily tell what is the "intelligent" pixel dimensions. Possibly the best we can do is clarify that what we meant as `video:shape`.

**Comment by darrenwiens** [comment]
Sure, would a property like `video:letterbox` with a value like `[left pixels, bottom pixels, right pixels, top pixels]` work?

**Reply by bradh** [reply]
I think it would be a useful option. Maybe a top corner offset + width, height, so you can just read off the size of the useful bits. Not entirely sure how easy it will be to determine without out-of-band information, but certainly a candidate.

**Reply by darrenwiens** [reply]
That sounds good. I'll follow the [rasterio window](https://rasterio.readthedocs.io/en/latest/topics/windowed-rw.html#windows) convention: `[col_off, row_off, width, height]`

**Reply by darrenwiens** [reply]
Addressed in: https://github.com/sparkgeo/stac-video/pull/12
